src/starlite_saqlalchemy/dto.py

```diff
@@ -152,7 +227,108 @@ class User(DeclarativeBase):
+    overridden on the decorated class).
+
+    Pydantic validation is supported using `validator` decorator from `starlite_sqlalchemy.pydantic`
+
+    As for the `factory` function, included fields can be controlled on the SQLAlchemy class
+    definition by including a "dto" key in the `Column.info` mapping.
+
+    Example:
+    ```python
+    class User(DeclarativeBase):
+        id: Mapped[UUID] = mapped_column(
+            default=uuid4, primary_key=True, info={"dto": Attrib(mark=dto.Mark.READ_ONLY)}
+        )
+        email: Mapped[str]
+        password_hash: Mapped[str] = mapped_column(info={"dto": Attrib(mark=dto.Mark.SKIP)})
+
+
+    @dto.dto(User, Purpose.WRITE)
+    class UserCreate:
+        @dto.validator("email")
+        def val_email(cls, v):
+            if "@" not in v:
+                raise ValueError("Invalid email")
+    ```
+
+    When setting `mapper_bind` (default to `True`), the resulting pydantic model will have a `mapper` method
+    that can be used to create an instance of the original SQLAlchemy model with values from the pydantic one:
+
+    ```python
+    @dto.dto(User, Purpose.WRITE)
+    class UserCreate:
+        pass
+
+
+    user = UserCreate(email="john@email.me")
+    user_model = user.mapper()  # user_model is a User instance
+    ```
+
+    Args:
+        model: The SQLAlchemy model class.
+        purpose: Is the DTO for write or read operations?
+        exclude: Explicitly exclude attributes from the DTO.
+        mapper_bind: Add a `mapper()` method that return an instance of the original SQLAlchemy model with values from the pydantic model.
+    Returns:
+        A Pydantic model that includes only fields that are appropriate to `purpose` and not in
+        `exclude`, except relationship fields.
+    """
+
+    def wrapper(cls: type) -> type[BaseModel]:
+        def wrapped() -> type[BaseModel]:
+            exclude_ = exclude or set[str]()
+            mapper = cast("Mapper", inspect(model))
+            columns = mapper.columns
+            relationships = mapper.relationships
+            fields: dict[str, tuple[Any, FieldInfo]] = {}
+            name = cls.__name__
+            namespace = {
+                **sys.modules[model.__module__].__dict__,
+                **sys.modules[cls.__module__].__dict__,
+            }
+            type_hints = {
+                **get_type_hints(model, localns=namespace),
+                **cls.__annotations__,
+            }
+            for key, type_hint in type_hints.items():
+                elem: Column | RelationshipProperty
+                try:
+                    elem = columns[key]
+                except KeyError:
+                    elem = relationships[key]
+                attrib = _get_dto_attrib(elem)
+                if _should_exclude_field(purpose, elem, exclude_, attrib):
+                    continue
+                type_ = type_hint
+                if key in relationships and key not in cls.__annotations__:
```

Comment:
What is the reasoning behind this bit?

Reply:
I think it was for testing, but it should be recursive by default as `factory` is.

Reply:
Oh, I remember now. The recursive implementation we have currently does not handle more complex case (that I'm facing) like `list[Model]`, `Optional[list[Model]]`, `list[A] | list[B]` etc. I think we should remove it untill we get something more robust.